java/custom/javax/microedition/lcdui/Display.jpp

```diff
@@ -3666,6 +3670,13 @@ public class Display {
                 }
             }
             // #endif ENABLE_CHAMELEON
+
+            KeyboardVisibilityListener listener = Display.this.getKeyboardVisibilityListener();
```

Comment:
We could make com.nokia.mid.ui.VirtualKeyboard listen for this event, so we wouldn't need to modify Display.jpp.